hail/python/test/hailtop/test_aiogoogle.py

```diff
@@ -248,3 +263,24 @@ async def listfiles(dir, recursive):
         else:
             stat = await entry.status()
             assert await stat.size() == 0
+
+@pytest.mark.asyncio
+async def test_multi_part_create(local_filesystem):
+    fs, base = local_filesystem
+
+    num_parts = 3
+    parts_data = [secrets.token_bytes(8192) for _ in range(num_parts)]
+
+    path = f'{base}a'
+    async with await fs.multi_part_create(path, num_parts) as c:
+        async def create_part(i):
+            async with await c.create_part(i, i * 8192) as f:
+                await f.write(parts_data[i])
+
+        await asyncio.gather(*[
```

Comment:
Can you randomize the order you write the parts or do it in reverse order just to make sure it's working?

Reply:
Done!